src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+export function getFilteredTodo(
+  todos: Todo[],
+  filterType: string,
+  query: string,
+) {
+  const filterBy = todos.filter((todo) => {
+    switch (filterType) {
+      case 'all':
+        return todo;
+
+      case 'completed':
+        return todo.completed === true;
+
+      case 'active':
+        return todo.completed === false;
+
+      default:
+        return 0;
+    }
+  });
+
+  return filterBy.filter(({ title }) => (
+    title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+  ));
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedTodo, setSelectedTodo] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+  const [filteredBy, setFilteredBy] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(todo => setTodos(todo))
```

Comment:
Try to `setIsLoading(false)` in `then` maybe you have soma delay because of finally, and this test fall.
Let's check.
<img width="765" alt="Screenshot 2022-09-28 at 12 35 56" src="https://user-images.githubusercontent.com/17098736/192745495-2519619a-871f-4368-9835-5767cf162874.png">